docs/source/_ext/directives.py

```diff
@@ -0,0 +1,89 @@
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives.tables import CSVTable
+from docutils import Component
+
+from sphinx.directives.code import CodeBlock
+
+from ChildProject.projects import ChildProject
+from ChildProject.annotations import AnnotationManager
+
+import os
+import pandas as pd
+from functools import reduce
+import subprocess
+import textwrap
+
+# I think this is a sort of hack (smart hack, but a hack nonetheless !)
+# and that it should not be done this way
+
+class CliDoc(CodeBlock):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        command_string = "\n".join(self.content).strip()
+        proc = subprocess.Popen(command_string, stdout = subprocess.PIPE, stderr = None, shell = True)
+        out = proc.communicate()[0]
+
+        if proc.returncode != 0:
+            raise Exception("CliDoc('{}') returned non-zero status code.".format(command_string))
+
+        if not self.arguments:
+            self.arguments = ['bash']
+
+        self.content = ["\n$ {}\n{}".format(command_string, out.decode("utf-8"))]
+
+class IndexTable(CSVTable):
```

Comment:
It would be great if someone with a decent knowledge of Sphinx/docutils could rework this class properly.